src/botPage/view/Dialogs/Limits.js

```diff
@@ -3,33 +3,51 @@ import PropTypes from 'prop-types';
 import { translate } from '../../../common/i18n';
 import * as style from '../style';
 import Dialog from './Dialog';
+import { restrictInputCharacter } from '../shared';
 
 class LimitsContent extends PureComponent {
     constructor() {
         super();
         this.state = {
-            error: null,
+            error    : '',
+            maxLoss  : '',
+            maxTrades: '',
         };
     }
     submit() {
-        const maxLoss = +this.maxLossDiv.value;
-        const maxTrades = +this.maxTradesDiv.value;
-        if (maxLoss > 0 && maxTrades > 0) {
-            if (maxTrades <= 100) {
-                this.props.onSave({
-                    maxLoss,
-                    maxTrades,
-                });
-            } else {
-                this.setState({
-                    error: translate('Maximum allowed number of trades for each session is 100.'),
-                });
-            }
-        } else {
-            this.setState({
-                error: translate('Both number of trades and loss amount have to be positive values.'),
-            });
+        let { maxLoss } = this.state;
+        const maxTrades = +this.state.maxTrades;
+        this.setState({ error: '' });
+        if (maxTrades <= 0 || maxTrades > 100) {
+            this.setState({ error: 'Max trade should be in range (1-100)' });
+            return;
         }
+        if (!maxLoss || !maxLoss.match(/^(?:\d*|\d*.\d\d?)$/) || maxLoss < 0.01) {
+            this.setState({ error: 'Max Loss should be a whole number or in multiples of 0.01' });
+            return;
+        }
+        maxLoss = +maxLoss;
+        this.props.onSave({
+            maxLoss,
+            maxTrades,
+        });
+    }
+    componentDidMount() {
+        const cleanupLayout = () => {
+            $(this.maxTradesDiv).val('');
```

Comment:
Now that you've added those values in states, you can reset the states instead of the UI elements.

Reply:
Done